### PR TITLE
2023 Updates

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Changes

* Removed .travis.yml
* Updated documentation building in Makefile
* Updated pre-commit hooks
* Updated dev dependencies
* Updated actions used for testing bake setup
* Added Issue Templates
* Added GitHub Workflows to the recipe
* Added a Zenodo configuration
* Updated license identifiers to use SPDX conventions.
* Fix some issues with spacing.

### What is not changed (but should be)

* We should eventually adopt `pyproject.toml` (with `flit`) in this recipe and dropping `setup.py`

### Other comments

The engine being used to test all this is largely unmaintained, hence why I needed to reduce the number of builds. From what I can see, generating (baking) the recipe works fine from most Python versions, but the testing utilities are not compatible with newer Python versions. Oh well.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -12,22 +12,20 @@ jobs:
...
         include:
-          - tox-env: py37
-            python-version: "3.7"
           - tox-env: py38
             python-version: "3.8"
           - tox-env: py39
             python-version: "3.9"
           - tox-env: pypy3
-            python-version: pypy3
+            python-version: "pypy3.9"
```
Comment (tlvu) [inline .github/workflows/main.yml:20]: This .github/workflows/main.yml is to test this cookiecutter only, not the generated project-slug right?  So do we even need all the different versions of python?

File: {{cookiecutter.project_slug}}/CONTRIBUTING.rst [hunk 2]
```diff
@@ -119,9 +119,7 @@ Before you submit a pull request, check that it meets these guidelines:
 2. If the pull request adds functionality, the docs should be updated. Put
    your new functionality into a function with a docstring, and add the
    feature to the list in README.rst.
-3. The pull request should work for Python 3.8, 3.9, and 3.10. Check
-   https://travis-ci.com/{{ cookiecutter.github_username }}/{{ cookiecutter.project_slug }}/pull_requests
-   and make sure that the tests pass for all supported Python versions.
+3. The pull request should work for Python 3.8, 3.9, 3.10, and 3.11. Check that the tests pass for all supported Python versions.
```
Comment (tlvu) [inline {{cookiecutter.project_slug}}/CONTRIBUTING.rst:122]: {{cookiecutter.project_slug}}/.github/workflows/main.yml only test up to 3.10, here we mention 3.11?

File: {{cookiecutter.project_slug}}/MANIFEST.in [hunk 3]
```diff
@@ -21,6 +23,9 @@ exclude .editorconfig
 exclude .pre-commit-config.yaml
 exclude .readthedocs.yml
 exclude .yamllint.yaml
+{% if cookiecutter.open_source_license != 'Not open source' -%}
+include .zenodo.json
```
Comment (tlvu) [inline {{cookiecutter.project_slug}}/MANIFEST.in:27]: An include among all the excludes?  Should be group all includes and excludes together?

File: {{cookiecutter.project_slug}}/Makefile [hunk 4]
```diff
@@ -36,6 +36,14 @@ clean-build: ## remove build artifacts
 	find . -name '*.egg-info' -exec rm -fr {} +
 	find . -name '*.egg' -exec rm -f {} +
 
+{%- if cookiecutter.make_docs == 'y' %}
+
+clean-docs: ## remove docs artifacts
+	rm -f docs/apidoc/ravenpy*.rst
```
Comment (tlvu) [inline {{cookiecutter.project_slug}}/Makefile:42]: hardcode of ravenpy name.  Should this be the project_slug?

File: {{cookiecutter.project_slug}}/Makefile [hunk 5]
```diff
@@ -79,24 +88,28 @@ coverage: ## check code coverage quickly with the default Python
 	$(BROWSER) htmlcov/index.html
 
 {%- if cookiecutter.make_docs == 'y' %}
-docs: ## generate Sphinx HTML documentation, including API docs
-	rm -f docs/{{ cookiecutter.project_slug }}.rst
-	rm -f docs/modules.rst
-	sphinx-apidoc -o docs/ {{ cookiecutter.project_slug }}
-	$(MAKE) -C docs clean
+autodoc: clean-docs ## create sphinx-apidoc files:
+	sphinx-apidoc -o docs/apidoc --private --module-first ravenpy
```
Comment (tlvu) [inline {{cookiecutter.project_slug}}/Makefile:92]: ravenpy hardcode, wonder where the template comes from hihi

File: {{cookiecutter.project_slug}}/setup.py [hunk 6]
```diff
@@ -45,6 +45,7 @@
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
```
Comment (tlvu) [inline {{cookiecutter.project_slug}}/setup.py:48]: Oh we mention 3.11 here as well ... I'll let you decide how to harmonize declaration with what is really tested.

File: {{cookiecutter.project_slug}}/.github/workflows/main.yml [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - tox-env: "py38"
+            python-version: "3.8"
+          - tox-env: "py39"
+            python-version: "3.9"
+          - tox-env: "py310"
+            python-version: "3.10"
```
Comment (tlvu) [inline {{cookiecutter.project_slug}}/.github/workflows/main.yml:51]: Here is the 3.10 I saw

File: {{cookiecutter.project_slug}}/tox.ini [hunk 8]
```diff
@@ -1,44 +1,55 @@
...
-
-[travis]
-python =
-    {% if cookiecutter.make_docs ==  'y' -%}
-    3.8: docs
-    {%- endif %}
+min_version = 4.0
+envlist =
+    black
+    py{38,39,310,311}
```
Comment (tlvu) [inline {{cookiecutter.project_slug}}/tox.ini:5]: here we have 3.11

**Review by tlvu — APPROVED** [review]
Very minor comments.

File: .github/workflows/main.yml (hunk 1 above)
Comment (Zeitsperre) [reply .github/workflows/main.yml:20]: I suppose not, but it's free, so why not.

File: {{cookiecutter.project_slug}}/MANIFEST.in (hunk 3 above)
Comment (Zeitsperre) [reply {{cookiecutter.project_slug}}/MANIFEST.in:27]: Good catch!

**Pull request merged**